### ISSUE TITLE
makes wrong suggestions

### ISSUE THREAD

**Issue by david-conecta-wireless** [microsoft/pyright#1700]
In the example the assignment of 4127 to knights ['guido'] marks it as error, however when executing the script, you can see in the output that the assignment is correct, python 3.6.8

![captura](https://user-images.githubusercontent.com/50668669/112961993-318d1580-9146-11eb-81c0-bc819d7a2466.png)

![captura1](https://user-images.githubusercontent.com/50668669/112962468-b4ae6b80-9146-11eb-8616-22ca5b93fe95.png)

**Comment by erictraut** [comment]
Pyright is a static type checker, and it's telling you that you have a type inconsistency. You have provided no type annotation for the variable `knights`, so it is using the assigned expression to infer that its value is `Dict[str, str]`. You are then trying to assign an `int` value to it, which violates its type. If your intent is for the dict to accept keys that are types other than `str`, you will need to provide the appropriate type annotation, such as:

```python
knights: Dict[str, Union[str, int]] = ...
```
or 
```python
knights: Dict[str, Any] = ...
```

For more details, refer to the documentation [here](https://github.com/microsoft/pyright/blob/master/docs/type-concepts.md) and [here](https://github.com/microsoft/pyright/blob/master/docs/type-inference.md).

If you are not interested in static type checking and want only language service features like completion suggestions, you can disable the type checking diagnostics by setting `python.analysis.typeCheckingMode` to `"off"`.

**Comment by erictraut** [comment]
If you have any further questions about type checking, the "Discussions" section of this repo is a good place to post.

If you have future bug reports, please include a self-contained piece of code (in text form) that exhibits the problem. Screen shots often do not contain sufficient context and make it difficult to repro the issue.
